In [28]:
import nsfg
import math
df = nsfg.ReadFemPreg()


In [29]:
preg = nsfg.ReadFemPreg()
live = preg[preg.outcome == 1]


In [30]:
firsts = live[live.birthord == 1]
others = live[live.birthord != 1]

In [31]:
firsts_weights = firsts.totalwgt_lb
others_weights = others.totalwgt_lb

Calculate Cohen's D for the total weight (in lbs) for the two groups, 'firsts' and 'others'.

In [35]:
diff = firsts_weights.mean() - others_weights.mean()
var1, var2 = firsts_weights.var(), others_weights.var()
n1, n2 = len(firsts_weights), len(others_weights)
pooled_var = (n1*var1 + n2*var2)/(n1+n2)
d=diff/math.sqrt(pooled_var)
d

-0.088672927072602

Cohen's d for this example is equal to -0.0887, which tells us that the difference in means is very small, and likely not clinically significant. The effect size of the difference of pregnancy lengths between firsts and others is also very small. 
